plutus-core/plutus-ir/Language/PlutusIR/TypeCheck/Internal.hs

```diff
@@ -287,25 +290,25 @@ checkKindFromBinding = \case
 checkTypeFromBinding(G,b)
 -}
 checkTypeFromBinding :: forall e uni a. (GShow uni, GEq uni, DefaultUni <: uni, AsTypeError e (Term TyName Name uni ()) uni a, AsTypeErrorExt e uni a)
-                 => Recursivity ->Binding TyName Name uni a  -> TypeCheckM uni e ()
+                 => Recursivity -> Binding TyName Name uni a  -> TypeCheckM uni IsTop e ()
 checkTypeFromBinding recurs = \case
     TypeBind{} -> pure () -- no types to check
     TermBind _ _ (VarDecl ann _ ty) rhs ->
-        checkTypeM ann rhs . fmap void =<< normalizeTypeM ty
+        noEscapeType (checkTypeM ann rhs . fmap void =<< normalizeTypeM ty)
```

Comment:
We need a note about how this changes as we go into let bindings, which we can reference from here.